### PR TITLE
Add spatie media library to demo

### PR DESCRIPTION
**Pull request by pxpm**
### BEFORE - What was wrong? What was happening before this PR?

No working examples on how to use Spatie Media Library with our fields. 

Installation steps:
1) You should do a `composer update` first to get media library installed, I already added it to `composer.json`
2) There is a new disk on `filesystems.php` along with links. If you are caching your configs, make sure to `php artisan config:clear`. 
3) Migrations are added/modified, so do a `php artisan migrate:fresh --seed`
4) setup the folder link: `php artisan storage:link`

You should be good to go. Navigate to Products Crud, you should see something like:
image

### AFTER - What is happening after this PR?

We should now evaluate the steps taken to use Spatie Media Library and brainstorm a way to make developers life easier. I see there is a lot of "repetitive" work that we can automate.

### CODE REVIEW

File: app/Models/Product.php
Comment (tabacitu) [inline app/Models/Product.php:133]: Let's do this as events in the CrudController, in Backpack itself.

```php
CRUD::field('specifications')->type('upload_multiple')->events([
    'saving' => function ($entry) {
        $entry->uploadMediaToCollection($media_name, $collection_name, $disk, $filename);
    },
    'retrieved' => function ($entry) {
        $entry->specifications = $entry->getMediaFromCollection($media_name, $collection_name, $disk);
    },
]);
```

Comment (tabacitu) [reply app/Models/Product.php:133]: Note: we'd still need 3 methods (not really with these names):
    => uploadImageToCollection()
    => uploadMediaToCollection()
    => uploadMultipleMediaToCollection()

They can live in CrudTrait, no problem. We just don't call them in the Model accessor/mutator, to not take over the Model.

Comment (tabacitu) [reply app/Models/Product.php:133]: If we move this into a `backpack/spatie-media-library-events` package we would:
- PRO - be able to tie that package to `spatie/media-library` package
- PRO - see download stats
- CON - have to maintain an extra micro-package;

**Comment by pxpm** [comment]
Note: we should review the final syntax so that I can update this branch with the correct definitions to work in demo.

**Comment by tabacitu** [comment]
Let's finish this PR to use the latest media syntax 💪

**Comment by pxpm** [comment]
@tabacitu testing notes.

This requires: https://github.com/Laravel-Backpack/CRUD/pull/4921

Everything should be configured in this PR, just don't forget to `php artisan migrate`, and clear configs in case you are caching them: `php artisan config:clear`. 

This should work out-of-the-box! 🙏

**Comment by tabacitu** [comment]
I've just gone through this PR and "refreshed" it:
- pointed PR to `v6` instead of `main`
- merged `v6` into it
- change the syntax etc, because the package had suffered some changes since it was first submitted;
- ran `composer update`

And I can confirm it is working WONDERFULLY:
CleanShot 2023-05-31 at 15 12 09@2x

There's only one thing I didn't do, that's preventing me from merging this is. We don't have a stopping mechanism for the online demo 👀 @pxpm could you please prevent people from _actually_ uploading, when this is in production? We can probably bind/intercept all uploaders in production, to prevent the uploads... right?! 🤷‍♂️

**Comment by pxpm** [comment]
Thanks for moving this foward, was my next on line 🙏 

> We can probably bind/intercept all uploaders in production, to prevent the uploads... right?!

Will investigate the best course of action here.

**Comment by pxpm** [comment]
This is pending https://github.com/Laravel-Backpack/medialibrary-uploaders/pull/17

As soon as that gets merged I think we are good to go here.

@tabacitu can you give this a double check in the production constrains to don't upload files ? 

Cheers

File: config/filesystems.php
Comment (tabacitu) [inline config/filesystems.php:70]:
```suggestion
```

File: config/filesystems.php
Comment (tabacitu) [inline config/filesystems.php:92]:
```suggestion
```

File: app/Http/Controllers/Admin/ProductCrudController.php
Comment (tabacitu) [reply app/Http/Controllers/Admin/ProductCrudController.php:302]: Not urgent - just a thought. Right now I just want to merge this ASAP.

**Review by tabacitu — CHANGES_REQUESTED** [review]
For UX and another layer of protection, I've made the upload fields disabled in production. For the repeatable I've chosen to hide it entirely.

I only want one extra thing out of this - for the mutator overrides to actually work, to prevent uploads instead of triggering an error.

I've added other ideas here, but those aren't urgent.

File: app/Http/Controllers/Admin/ProductCrudController.php
Comment (pxpm) [reply app/Http/Controllers/Admin/ProductCrudController.php:302]: Are you thinking something like this:
```php
if (! env('APP_ENV') === 'production') {
    trait DropzoneOperation
    {
        public function dropzoneUpload()
        {
            return response()->json([]);
        }
    }
} else {
    trait DropzoneOperation
    {
        use \Backpack\Pro\Http\Controllers\Operations\DropzoneOperation;
    }
}
```

And then in demo we use `App\Http\Controllers\Operations\DropzoneOperation` instead of the one in the package ? 

If you agree I can push this change. 

Cheers

File: app/Models/Product.php
Comment (pxpm) [reply app/Models/Product.php:84]: We can catch the `withMedia` and `withFiles`, with that we prevent the uploader events from beeing registered, so we wouldn't even need to bind into the uploader classes as they wouldn't be registered/called. 

That wouldn't prevent much, just the upload from happening, but anything you submit in the input would be saved to the database, except it didn't go through the upload process.

At the moment we can't bind to our uploaders as they aren't resolved from the container, they are statically initialized. 
We can work that out, or use the mutator as we are doing now. Everything is "working", except we always send empty values to the uploaders 🤷 

Probably it's like you said, it's more future proof, but at the moment it's not a requirement.

**Comment by tabacitu** [comment]
Tested, works, merging 🎉

Let's postpone the other 2 suggestions, they're COULDs/SHOULDs. Added issues for them for later in the year:
- https://github.com/Laravel-Backpack/demo/issues/528
- https://github.com/Laravel-Backpack/demo/issues/527

**Pull request merged**
